dora/core/server/worker/src/test/java/alluxio/worker/dora/DoraMetaManagerTest.java

```diff
@@ -0,0 +1,154 @@
+    UnderFileSystem system = mock(UnderFileSystem.class);
+    doThrow(new IOException()).when(system).listStatus(anyString(), any());
+    doReturn(system).when(mDoraUfsManager).getOrAdd(any(), any());
+
+    assertThrows(IOException.class, () -> {
+      mManager.listFromUfs("/test", false);
+    });
+  }
+
+  @Test
+  public void testListFromUfsGetWhenNull() throws IOException {
+    UnderFileSystem system = mock(UnderFileSystem.class);
+    when(system.listStatus(anyString())).thenReturn(null);
+    when(system.getStatus(anyString())).thenReturn(null);
+    doReturn(system).when(mDoraUfsManager).getOrAdd(any(), any());
+
+    Optional<UfsStatus[]> status = mManager.listFromUfs("/test", false);
+    assertEquals(status, Optional.empty());
+  }
+
+  @Test
+  public void testListFromUfsGetWhenFail() throws IOException {
+    UnderFileSystem system = mock(UnderFileSystem.class);
+    when(system.listStatus(anyString())).thenReturn(null);
+    when(system.getStatus(anyString())).thenThrow(new FileNotFoundException());
+    doReturn(system).when(mDoraUfsManager).getOrAdd(any(), any());
+
+    Optional<UfsStatus[]> status = mManager.listFromUfs("/test", false);
+    assertEquals(status, Optional.empty());
+  }
+
+  @Test
+  public void testListFromUfsGetWhenSuccess() throws IOException {
+    UnderFileSystem system = mock(UnderFileSystem.class);
+    UfsStatus fakeStatus = mock(UfsStatus.class);
+    when(system.listStatus(anyString())).thenReturn(null);
+    when(system.getStatus(anyString())).thenReturn(fakeStatus);
+    doReturn(system).when(mDoraUfsManager).getOrAdd(any(), any());
+
+    when(fakeStatus.getName()).thenReturn("test");
+
+    Optional<UfsStatus[]> status = mManager.listFromUfs("/test", false);
+    assertEquals(status.get()[0].getName(), "test");
+  }
+
+  @Test
+  public void testListFromUfsThenCacheWhenFail() throws IOException {
+    UnderFileSystem system = mock(UnderFileSystem.class);
+    doThrow(new IOException()).when(system).listStatus(anyString(), any());
+    doReturn(system).when(mDoraUfsManager).getOrAdd(any(), any());
+
+    assertThrows(IOException.class, () -> {
+      mManager.listFromUfsThenCache("/test", false);
+    });
+  }
+
+  @Test
+  public void testListFromUfsThenCacheWhenNull() throws IOException {
+    UnderFileSystem system = mock(UnderFileSystem.class);
+    when(system.listStatus(anyString())).thenReturn(null);
+    when(system.getStatus(anyString())).thenReturn(null);
+    doReturn(system).when(mDoraUfsManager).getOrAdd(any(), any());
+
+    Optional<UfsStatus[]> status = mManager.listFromUfsThenCache("/test", false);
+    assertEquals(status, Optional.empty());
+  }
+
+  private void deleteDirectory(File dir) {
+    File[] filelist = dir.listFiles();
+    if (filelist != null) {
+      for (File file : filelist) {
+        if (file.isDirectory()) {
+          deleteDirectory(file);
+        } else {
+          file.delete();
+        }
+      }
+    }
+    dir.delete();
+  }
```

Comment:
Thanks for your recursion function, it is elegant, but I think there is a method that recursively delete directories, why not use an existing one?